_source/_posts/2019-12-30-java-jersey-jaxrs.adoc

```diff
@@ -0,0 +1,595 @@
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.oauth2ResourceServer()
+                .and()
+                .authorizeRequests()
+                .anyRequest()
+                .authenticated();
+    }
+}
+----
+
+The configuration above will ensure all your requests will be authenticated. If you're using Spring MVC you don't need to add this configuration, but since you're developing with Jersey you need to make sure they are also included in the authentication process.
+
+Now that you added the library and the configuration, you can inform the following variables inside the `src/main/resources/application.properties` inside your project:
+
+[source,properties]
+----
+okta.oauth2.issuer: https://{yourOktaDomain}/oauth2/default
+----
+
+If you want to avoid adding this configuration to source control, you can use environment variables:
+
+[source,properties]
+----
+OKTA_OAUTH2_ISSUER=https://{yourOktaDomain}/oauth2/default
+----
+
+Your `{yourOktaDomain}` will be visible in your Okta dashboard, just click on the Dashboard on the menu. You will see the Org URL in the right upper corner.
+
+Now your application is secure!
+
+Let's try to make a request to one of your endpoints. Run your application with your latest changes, then go to your terminal line and execute the following command:
+
+[source,text]
+----
+curl -X GET -I http://localhost:8080/tasks
+----
+
+The result should be similar to this one:
+
+[source,text]
+----
+HTTP/1.1 401
+Set-Cookie: JSESSIONID=06775BFFBFDB74DA632CB6F4D973ADA4; Path=/; HttpOnly
+WWW-Authenticate: Bearer
+X-Content-Type-Options: nosniff
+X-XSS-Protection: 1; mode=block
+Cache-Control: no-cache, no-store, max-age=0, must-revalidate
+Pragma: no-cache
+Expires: 0
+X-Frame-Options: DENY
+Content-Type: text/html;charset=utf-8
+Content-Language: en
+Content-Length: 802
+Date: Mon, 04 Nov 2019 12:52:52 GMT
+----
+
+The status code of the response is `HTTP 401`, which means the request was not authorized to execute. In other words, your application is now secure! You need a valid token to make a request to your endpoints.
+
+Let's see how you can generate a valid token and how to add it to your request.
+
+== Generate a Valid Token
+
+To validate your request you need to add the `Authorization` header to the request. The header will provide the type of authentication and the token, which will look like the snippet below:
+
+[source,text]
+----
+-H 'Authorization: Bearer <token>'
+----
+
+To generate the token you can access https://oidcdebugger.com/ and provide the following information:
+
+- Authorize URI: `\https://{yourOktaDomain}/oauth2/default/v1/authorize`
+- Redirect URI: `\https://oidcdebugger.com/debug`
```

Comment:
Remove backslash from the above two links.

Reply:
I ran into an odd asciidoc issue, that caused an html tag escape problem, adding the backslashes stopped the href rendering, and the broken tag issue. Have you ever seen that?